zdm/tests/mc_statistics.py

```diff
@@ -0,0 +1,386 @@
+	
+	
+	# sets the nature of scaling with the 'spectral index' alpha
+	alpha_method=0 # spectral index interpretation: includes k-correction. Slower to update
+	#alpha_method=1 # rate interpretation: extra factor of (1+z)^alpha in source evolution
+	
+	############## Initialise surveys ##############
+	
+	# constants of intrinsic width distribution
+	Wlogmean=1.70267
+	Wlogsigma=0.899148
+	DMhalo=50
+	
+	#These surveys combine time-normalised and time-unnormalised samples 
+	NewSurveys=False
+	#sprefix='Full' # more detailed estimates. Takes more space and time
+	sprefix='Std' # faster - fine for max likelihood calculations, not as pretty
+	
+	if sprefix=='Full':
+		Wbins=10
+		Wscale=2
+		Nbeams=[20,20,20]
+	elif sprefix=='Std':
+		Wbins=5
+		Wscale=3.5
+		Nbeams=[5,5,10]
+	
+	# location for survey data
+	sdir = os.path.join(resource_filename('zdm', 'data'), 'Surveys/')
+	if NewSurveys:
+		
+		print("Generating new surveys, set NewSurveys=False to save time later")
+		#load the lat50 survey data
+		lat50=survey.survey()
+		lat50.process_survey_file(sdir+'CRAFT_class_I_and_II.dat')
+		lat50.init_DMEG(DMhalo)
+		lat50.init_beam(nbins=Nbeams[0],method=2,plot=False,thresh=thresh) # tells the survey to use the beam file
+		pwidths,pprobs=survey.make_widths(lat50,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+		efficiencies=lat50.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+		
+		
+		# load ICS data
+		ICS=survey.survey()
+		ICS.process_survey_file(sdir+'CRAFT_ICS.dat')
+		ICS.init_DMEG(DMhalo)
+		ICS.init_beam(nbins=Nbeams[1],method=2,plot=False,thresh=thresh) # tells the survey to use the beam file
+		pwidths,pprobs=survey.make_widths(ICS,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+		efficiencies=ICS.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+		
+		# load Parkes data
+		pks=survey.survey()
+		pks.process_survey_file(sdir+'parkes_mb_class_I_and_II.dat')
+		pks.init_DMEG(DMhalo)
+		pks.init_beam(nbins=Nbeams[2],method=2,plot=False,thresh=thresh) # need more bins for Parkes!
+		pwidths,pprobs=survey.make_widths(pks,Wlogmean,Wlogsigma,Wbins,scale=Wscale)
+		efficiencies=pks.get_efficiency_from_wlist(dmvals,pwidths,pprobs)
+		
+		
+		names=['ASKAP/FE','ASKAP/ICS','Parkes/Mb']
+		
+		surveys=[lat50,ICS,pks]
+		if not os.path.isdir('Pickle'):
+			os.mkdir('Pickle')
+		with open('Pickle/'+sprefix+'surveys.pkl', 'wb') as output:
+			pickle.dump(surveys, output, pickle.HIGHEST_PROTOCOL)
+			pickle.dump(names, output, pickle.HIGHEST_PROTOCOL)
+	else:
+		with open('Pickle/'+sprefix+'surveys.pkl', 'rb') as infile:
+			surveys=pickle.load(infile)
+			names=pickle.load(infile)
+			lat50=surveys[0]
+			ICS=surveys[1]
+			pks=surveys[2]
+	print("Initialised surveys ",names)
+	
+	dirnames=['ASKAP_FE','ASKAP_ICS','Parkes_Mb']
+	
+	#### these are hard-coded best-fit parameters ####
+	# initial parameter values. SHOULD BE LOGSIGMA 0.75! (WAS 0.25!?!?!?)
+	# Best-fit parameter values (result from cube iteration)
+	lEmin=30. # log10 in erg
```

Comment:
Would be best if these were input to the method so that
we can vary them in the future.

Reply:
As in lEmin is loaded at the input stage? Yes, agreed. But we would also want the pickling to be passed along. Probably the program should be: EITHER: load a pickled grid and extract params from there, OR: generate a new grid with given parameters.

Reply:
Since this is an extra feature, unrelated to this pull request, I'll leave this for the moment.